### ISSUE TITLE
unused-import false positive for names in string argument to typing.cast

### ISSUE THREAD

**Issue by lggruspe** [PyCQA/pylint#7548]
### Bug description

example.py:

```python
# pylint: disable=missing-docstring
from pathlib import Path
import typing as t

paths = t.cast("set[Path]", set())
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:2:0: W0611: Unused Path imported from pathlib (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

There shouldn't be any messages.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-1)]
```

### OS / Environment

fedora 36

**Comment by clavedeluna** [comment]
I'm able to reproduce. Worth noting, though, that even PyCharm registers `Path` as unused

**Comment by bersbersbers** [comment]
This false positive is prone to receiving much additional exposure now that `ruff` has implemented `TC006`, compare https://github.com/astral-sh/ruff/pull/14511, leading to the stringification of the type argument in practically all `cast` calls.

`TC006` is still in preview mode, but that won't have to remain that way. https://docs.astral.sh/ruff/rules/runtime-cast-value/

**Comment by Pierre-Sassoulas** [comment]
Nice catch @bersbersbers thank you. This check has to be activated in ruff so it's not going to be a flash flood, but I labelled as "need PR".
